buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -435,7 +443,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_oauth_clients_version = "1.31.0"
     // Try to keep grpc_version consistent with gRPC version in google_cloud_platform_libraries_bom
     def grpc_version = "1.32.2"
-    def guava_version = "25.1-jre"
+    def guava_version = guava25Projects.contains(project.path) ? "25.1-jre" : "30.1-jre"
```

Comment:
Would not this be problematic, causing Beam to depend on 2 different versions? Which version, users of Beam will be depending if they need to use Beam with one of these 3 projects?

Reply:
There's no impact to the Beam Cassandra and Hadoop artifacts. The Maven artifact `org.apache.beam:beam-sdks-java-io-hadoop-format:2.27.0`, `org.apache.beam:beam-sdks-java-io-cassandra:2.27.0`, or `org.apache.beam:beam-sdks-java-io-hadoop-file-system:2.27.0` does not declare Guava dependency.

# Instruction for Hadoop / Cassandra Beam users

If Beam Cassandra / Hadoop users use Beam with beam-sdks-java-io-kinesis, beam-sdks-java-io-google-cloud-platform, or beam-sdks-java-extensions-sql-zetasql (they declare Guava dependency), then the users need to pin Guava version to 25.1-jre. They can use `<dependencyManagement>` for Maven and `force` for Gradle.

Reply:
Ack. I think this would be an undocumented hurdle for the impacted users. I am not sure what is the best course of action. Hopefully @kennknowles would have a recommendation.

Reply:
Yes, I think I should document that condition ("_if Beam Cassandra / Hadoop users use Beam with beam-sdks-java-io-kinesis, ..._") somewhere.